examples/repo_example/basic_repo.py

```diff
@@ -0,0 +1,403 @@
+# ---------
+# Update top-level targets to delegate trust for specific file path
+# patterns to the new role, identified by name and key.
+roles["targets"].signed.delegations = Delegations(
+    keys={
+        keys[delegatee_name]["keyid"]: Key.from_securesystemslib_key(
+            keys[delegatee_name]
+        )
+    },
+    roles=OrderedDict(
+        [
+            (
+                delegatee_name,
+                DelegatedRole(
+                    name=delegatee_name,
+                    keyids=[keys[delegatee_name]["keyid"]],
+                    threshold=1,
+                    terminating=True,
+                    paths=["*.py"],
+                ),
+            )
+        ]
+    ),
+)
+
+
+# Remove target file info from top-level targets (this has been delegated)
+del roles["targets"].signed.targets[target_file_path]
+
+
+# Snapshot + Timestamp + Sign + Persist
+# -------------------------------------
+
+# Top-level targets metadata has changed, so we need to bump its version
+roles["targets"].signed.version += 1
+
+# Account for changed and new targets in snapshot
+roles["snapshot"].signed.meta["targets.json"].version = roles[
+    "targets"
+].signed.version
+roles["snapshot"].signed.meta[f"{delegatee_name}.json"] = MetaFile(version=1)
+roles["snapshot"].signed.version += 1
+
+# Account for new snapshot in timestamp
+roles["timestamp"].signed.snapshot_meta.version = roles[
+    "snapshot"
+].signed.version
+roles["timestamp"].signed.version += 1
+
+# Sign and write all changed files, i.e. all but root
+for role_name in ["targets", "python-scripts", "snapshot", "timestamp"]:
+    signer = SSlibSigner(keys[role_name])
+    roles[role_name].sign(signer)
+
+    # Prefix all but timestamp metadata with version number
+    filename = f"{role_name}.json"
+    if role_name != "timestamp":
+        filename = f"{roles[role_name].signed.version}.{filename}"
+
+    roles[role_name].to_file(os.path.join(TMP_DIR, filename), serializer=PRETTY)
+
+
+# Root key rotation (recover from a compromise / key loss)
+# =======================================================
+# TODO: Explain different types of key compromises (non-root top-level, and
+# delegated targets) and how root is the most severe and requires special care
+# (sign with new and old keys, be quicker than the attacker)
+
+# Create two new root keys
+new_root_keys = [generate_ed25519_key()] * 2
+
+
+# Revoke compromised root key
+roles["root"].signed.remove_key("root", keys["root"]["keyid"])
+
+
+# Add new keys
+for key in new_root_keys:
+    public_key = Key.from_securesystemslib_key(key)
+    roles["root"].signed.add_key("root", public_key)
+
+# Bump version
+roles["root"].signed.version += 1
+
+# Clear old signatures
+roles["root"].signatures.clear()
+
+# Sign with threshold of old keys and write
+# Just assume we do out-of-band signing for keys we don't have
+for key in [keys["root"], another_root_key] + new_root_keys:
+    roles["root"].sign(SSlibSigner(key), append=True)
+
```

Comment:
I think the part "signing with a threshold of old keys AND a threshold of the new keys" needs an emphasis.

Reply:
➕  I absolutely agree.